### PR TITLE
Check for ROMs at beginning of generation

### PR DESCRIPTION
**Pull request by espeon65536**
Games with ROM output used to only load the ROM when needed. Now we check for the existence of the file path at the beginning, which will catch failed generations much earlier.

### CODE REVIEW

File: Main.py [hunk 1]
```diff
@@ -93,6 +93,8 @@ def main(args, seed=None, baked_server_options: Optional[Dict[str, object]] = No
                         f"Location IDs: {min(cls.location_id_to_name):{numlength}} - "
                         f"{max(cls.location_id_to_name):{numlength}}")
 
+    AutoWorld.call_stage(world, "can_generate")
```
Comment (JarnoWesthof) [inline Main.py:96]: i would expect a can_ method to return a boolean rather than raise an exception

**Review by JarnoWesthof — CHANGES_REQUESTED** [review]

File: Main.py (hunk 1 above)
Comment (Berserker66) [reply Main.py:96]: yeah, this might be more of an assert_generate or raises_generate_issues

Comment (espeon65536) [reply Main.py:96]: Ok, will change to `assert_generate`

File: worlds/oot/__init__.py [hunk 2]
```diff
@@ -89,6 +89,11 @@ def __init__(self, world, player):
         self.hint_data_available = threading.Event()
         super(OOTWorld, self).__init__(world, player)
 
+    @classmethod
+    def stage_assert_generate(cls, world: MultiWorld):
+        rom = Rom(file=get_options()['oot_options']['rom_file'])
+        del rom
```
[the remark this answers is not in the document]
Comment (espeon65536) [reply worlds/oot/__init__.py:95]: I know, the del is intentional to make intent clear. If you'd rather not have them for style reasons then I can remove them.

[the remark this answers is not in the document]
Comment (Berserker66) [reply worlds/oot/__init__.py:95]: I'd prefer nothing or a comment, over the del, personally.

**Comment by black-sliver** [comment]
Would you mind adding this to docs/api.md?

**Comment by Berserker66** [comment]
also there's still a merge conflict in case you didn't see the notice.

**Review by KonoTyran — APPROVED** [review]

**Pull request merged**